test_integration/erc20_gateway/03_withdraw_and_confirm_withdraw/03_withdraw_and_confirm_withdraw.ts

```diff
@@ -0,0 +1,194 @@
+      withdrawParams.beneficiary,
+      withdrawParams.feeGasPrice.toString(10),
+      withdrawParams.feeGasLimit.toString(10),
+      withdrawParams.utilityToken,
+    );
+
+    const tx = await Utils.sendTransaction(
+      withdrawRawTx,
+      {
+        from: withdrawerAddress,
+      }
+    );
+
+    withdrawMessageHash = tx.events.WithdrawIntentDeclared.returnValues['messageHash'];
+
+    const withdrawerBalanceAfterWithdraw = await utilityTokenInstance.methods.balanceOf(
+      withdrawerAddress,
+    ).call();
+
+    await Assert.assertWithdrawIntentDeclared(
+      tx.events.WithdrawIntentDeclared,
+      withdrawerAddress,
+      withdrawParams,
+    );
+
+    Assert.assertWithdraw(
+      new BN(withdrawerBalanceBeforeWithdraw),
+      new BN(withdrawerBalanceAfterWithdraw),
+      withdrawParams.withdrawalAmount,
+    );
+  });
+
+  it('should anchor successfully', async (): Promise<void> => {
+    const anchor = await Utils.performAnchor(
+      shared.contracts.OriginAnchor.instance,
+      shared.consensus,
+    );
+
+    Assert.assertAnchor(
+      anchor.tx.events.StateRootAvailable,
+      anchor.blockNumber,
+      anchor.stateRoot,
+    );
+
+    blockNumber = anchor.blockNumber;
+  });
+
+  it('should prove ERC20Gateway successfully', async(): Promise<void> => {
+    const proof = await Utils.getAccountProof(
+      erc20Cogateway.address,
+      blockNumber.toString(10),
+    );
+
+    const rawTx = erc20Gateway.instance.methods.proveGateway(
+      blockNumber.toString(10),
+      // @ts-ignore
+      proof.encodedAccountValue,
+      // @ts-ignore
+      proof.serializedProof,
+    );
+
+    const tx = await Utils.sendTransaction(
+      rawTx,
+      {
+        from: shared.facilitator,
+      }
+    );
+
+    Assert.assertGatewayProven(
+      tx.events.GatewayProven,
+      erc20Cogateway.address,
+      new BN(blockNumber),
+    );
+  });
+
+  it('should confirm withdraw successfully', async (): Promise<void> => {
+    const outboxStorageIndex = await erc20Gateway.instance.methods.outboxStorageIndex().call();
+    const storagePath = Utils.storagePath(outboxStorageIndex, [withdrawMessageHash]);
+
+    const serializedStorageProof = await Utils.getStorageProof(
+      erc20Cogateway.address,
+      [storagePath],
+      blockNumber.toString(10),
+    );
+
+    const facilitatorBalanceBeforeConfirmWithdraw = await shared.contracts.ValueToken.instance.methods.
+      balanceOf(shared.facilitator).call();
+
+    const beneficiaryBalanceBeforeConfirmWithdraw = await shared.contracts.ValueToken.instance.methods.
+      balanceOf(withdrawParams.beneficiary).call();
+
+    const rawTx = erc20Gateway.instance.methods.confirmWithdraw(
+      utilityToken,
+      shared.contracts.ValueToken.address,
+      withdrawParams.withdrawalAmount,
+      withdrawParams.beneficiary,
+      withdrawParams.feeGasPrice,
+      withdrawParams.feeGasLimit,
+      shared.beneficiaryAtAuxiliary, // withdrawer address
```

Comment:
Can this be kept with withdrawParams instead of shared.

Reply:
Removed from shared and declared globally in the required integration test files.